.github/workflows/main.yml

```diff
@@ -0,0 +1,40 @@
+name: Main
+
+on:
+  push:
+  pull_request:
+    branches: [ master ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  Rust-Tests:
+
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: thavalon-server
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      run: cargo build --verbose
+    - name: Run tests
+      run: cargo test --verbose
```

Comment:
We could also use https://github.com/actions-rs/cargo for this - it has some nice features especially if we need a nightly toolchain.

Also, GitHub actions support caching dependencies, which will speed this up a fair amount - https://github.com/bnavetta/samwise/blob/master/.github/workflows/build.yml#L46-L55

Reply:
@theadd336  to address